src/modules/projectContext.js

```diff
@@ -1,82 +1,55 @@
-import { TraceContext } from './traceContext.js';
+import { AbstractProjectContext } from './abstractProjectContext.js';
+import { SessionManager } from './sessionManager.js';
 
-class ProjectContext {
+class ProjectContext extends AbstractProjectContext {
 
     /**
      * Create a context for a specific project. This can be used to search in a project or list its images.
      *
      * @param {SessionManager} sessionManager The session manager, used for connections to the Hansken servers
-     * @param {'projects' | 'singlefiles'} collection 'projects' or 'singlefiles'
      * @param {UUID} collectionId The project id or single file id
      */
-    constructor(sessionManager, collection, collectionId) {
-        this.sessionManager = sessionManager;
-        this.collection = collection;
-        this.collectionId = collectionId;
+    constructor(sessionManager, collectionId) {
+        super(sessionManager, 'projects', collectionId);
     }
 
     /**
-     * Delete the project or singlefile.
+     * Create an image and link it to the project.
+     * This method should not be used with singlefiles.
      *
-     * @returns A promise
+     * @param {object} image The image as specified in the REST API docs.
+     * @returns The image id
      */
-    delete = () => this.sessionManager.gatekeeper(`/${this.collection}/${this.collectionId}`, {
-        method: 'DELETE'
-    });
-
-    /**
-     * Get the project or singlefile.
-     *
-     * @returns The project
-     */
-    get = () => this.sessionManager.gatekeeper(`/${this.collection}/${this.collectionId}`).then(this.sessionManager.toJson);
-
-    /**
-     * Update an existing project or singlefile.
-     *
-     * @param {object} project The new project to store for the specified project
-     * @returns A promise
-     */
-    update = (project) => this.sessionManager.gatekeeper(`/${this.collection}/${this.collectionId}`, {
-        method: 'PUT',
+    createImage = (image) => this.sessionManager.gatekeeper('/images', {
+        method: 'POST',
         headers: {
             'Content-Type': 'application/json'
         },
-        body: JSON.stringify(project)
-    });
+        body: JSON.stringify(image)
+    }).then(SessionManager.parseLocationId)
+    .then(this.linkImage);
 
     /**
-     * Get all images linked to a specific project or singlefile.
+     * Link an image to this project.
+     * This method should not be used with singlefiles.
      *
-     * @returns An array of images linked to this project
+     * @param {UUID} imageId The image id
+     * @returns The new ProjectImageContext
      */
-    // NOTE: /singlefiles/{singlefileId}/images does not exist, but one can use /projects/{singlefileId}/images
-    images = () => this.sessionManager.gatekeeper(`/projects/${this.collectionId}/images`).then(this.sessionManager.toJson);
+    linkImage = (imageId) => this.sessionManager.gatekeeper(`/projects/${this.collectionId}/images/${imageId}`, {
```

Comment:
I would rather see `projectId` here, but I get that it might be a bit hard to combine this with code sharing.

Reply:
True